src/main/java/com/intuit/graphql/orchestrator/batch/EntityFetcherBatchResultTransformer.java

```diff
@@ -64,6 +48,51 @@ public List<DataFetcherResult<Object>> toBatchResult(DataFetcherResult<Map<Strin
                     .build();
         }
 
+        mapErrorsToEntities(dataFetcherResult, _entities);
+
+        //using IntStream instead of regular for loop or atomic reference, so we can parallelize this if we want
+        IntStream.range(0, _entities.size()).forEach( idx -> {
+            Map<String, Object> entityInfo = _entities.get(idx);
+            Object fieldData = (entityInfo != null) ? entityInfo.get(this.extFieldName) : null;
+            List<GraphQLError> graphQLErrors = (List<GraphQLError>) entityInfo.getOrDefault("errors", new ArrayList<>());
+
+            dataFetcherResults.add(DataFetcherResult.newResult()
+                    .data(fieldData)
+                    .errors(graphQLErrors)
+                    .build()
+            );
+        });
+
         return dataFetcherResults;
     }
+
+    private void mapErrorsToEntities( DataFetcherResult<Map<String, Object>> dataFetcherResult, List<Map<String, Object>> entityList) {
+        if(dataFetcherResult.hasErrors()) {
+            dataFetcherResult.getErrors().forEach(error -> {
+                error.getExtensions().put("serviceNamespace", providerNamespace);
+                error.getExtensions().put("parentTypename", entityName);
+                error.getExtensions().put("fieldName", extFieldName);
+                error.getExtensions().put("downstreamErrors", error.toSpecification());
+
+                //Decided to ignore if errors have no path to match field resolver. This should not happen though.
+                if(error.getPath() != null) {
+                    //path is always _entities/entityIdx/entityField for entity fetch
+                    Integer pathIndex = (Integer) error.getPath().get(1);
+                    Map<String, Object> entityInfo = entityList.get(pathIndex);
+
+                    if(entityInfo != null) {
+                       List<GraphQLError> errors = (List<GraphQLError>) entityInfo.get("errors");
```

Comment:
I'm not sure if the entity field contains errors?  I think it should be in dataFetcherResult.getErrors.